Sources/Rendering/SceneGraph/ViewNodeFactory/index.js

```diff
@@ -15,10 +15,24 @@ function vtkViewNodeFactory(publicAPI, model) {
     if (dataObject.isDeleted()) {
       return null;
     }
-    if (Object.keys(model.overrides).indexOf(dataObject.getClassName()) === -1) {
+    // if (Object.keys(model.overrides).indexOf(dataObject.getClassName()) === -1) {
+    //   return null;
+    // }
+
+    const hierarchy = dataObject.getHierarchy();
```

Comment:
I'm guessing the `.isA()` is not enough, and you need to check the class name hierarchy in reverse order?

That seems strange, and instead of providing a method to get the full hierarchy, I would maybe prefer to extend the existing method like follow:

`publicAPI.getClassName = (depth=0) => model.classHierarchy.slice(-1 - depth)[0];`

Reply:
yes, that's the point.
I understand the extension, but how do you know the depth ? If I try getClassName(10) and the object is a vtkProp, what does it do ?

Reply:
If you go beyond the bounds of the classHierarchy array, it will always return the 1st element which is vtkObject.

```
const handledClasses = Object.keys(model.overrides);
let className = '';
let classNameDepth = 0;
while (className !== 'vtkObject' && handledClasses.indexOf(className) === -1) {
    className = dataObject.getClassName(classNameDepth++);
}

if (className === 'vtkObject') {
   return null;
}
```

Reply:
I'm not sure to understand, do I have to add this last code ? Because I don't find it.

Reply:
The code I wrote is the replacement of your change below if you extend the `getClassName` the way I was previously suggesting.

```
 +    let isObject = false;
 +    let className = '';
 +    for (let i = hierarchy.length - 1; i > 0; i--) {
 +      if (Object.keys(model.overrides).indexOf(hierarchy[i]) !== -1) {
 +        isObject = true;
 +        className = hierarchy[i];
 +        break;
 +      }
 +    }
 +    if (!isObject) {
        return null;
      }
```

Reply:
Ok thanks !

Reply:
A better solution would be 

`publicAPI.getClassName = (depth=0) => model.classHierarchy[model.classHierarchy.length - 1 - depth];`

This will prevent array allocation and you will get `undefined` if you go out of bound.